package/samplers/bisect/_sampler.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import annotations
+
+import math
+from typing import TYPE_CHECKING
+
+import numpy as np
+import optuna
+from optuna.trial import TrialState
+
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from typing import Any
+
+    from optuna import Study
+    from optuna import Trial
+    from optuna.distributions import BaseDistribution
+    from optuna.distributions import FloatDistribution
+    from optuna.distributions import IntDistribution
+    from optuna.trial import FrozenTrial
+
+
+PREFIX_LEFT = "bisect:left_"
+PREFIX_RIGHT = "bisect:right_"
+
+
+class BisectSampler(optuna.samplers.BaseSampler):
+    """Sampler using bisect (binary search) indepedently for each parameter.
+
+    Args:
+        rtol:
+            The relative tolerance parameter to be used to judge whether all the parameters are
+            converged. Default to that in `np.isclose`, i.e., 1e-5.
+        atol:
+            The absolute tolerance parameter to be used to judge whether all the parameters are
+            converged. Default to that in `np.isclose`, i.e., 1e-8.
+    """
+
+    def __init__(self, *, rtol: float = 1e-5, atol: float = 1e-8) -> None:
+        self._atol = atol
+        self._rtol = rtol
+        self._search_space: dict[str, IntDistribution | FloatDistribution] = {}
+        self._stop_flag = False
+
+    def infer_relative_search_space(
+        self, study: optuna.Study, trial: optuna.trial.FrozenTrial
+    ) -> dict[str, optuna.distributions.BaseDistribution]:
```

Comment:
The addition of `optuna.distributions` is inconsistent. We can remove it because `BaseDistributions` is already imported.
```suggestion
    ) -> dict[str, BaseDistribution]:
```